File: HashTable/Examples/jewels_and_stones.py
# https://leetcode.com/explore/learn/card/hash-table/187/conclusion-hash-table/1136/


class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewelsSet = set()

        for jewel in jewels:
            jewelsSet.add(jewel)

        count = 0

        for stone in stones:
            if stone in jewelsSet:
                count += 1

        return count


# Test ----------------------------
jewels = "aA"
stones = "aAAbbbb"

sol = Solution()

print(sol.numJewelsInStones(jewels, stones))

File: HashTable/Examples/jewels_and_stones.cpp
// https://leetcode.com/explore/learn/card/hash-table/187/conclusion-hash-table/1136/

#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    int numJewelsInStones(string jewels, string stones)
    {
        unordered_set<char> jewelsSet;

        for (char c : jewels)
        {
            jewelsSet.insert(c);
        }

        int count = 0;

        for (char stone : stones)
        {
            if (jewelsSet.count(stone) > 0)
            {
                ++count;
            }
        }

        return count;
    }
};

int main()
{
    string jewels = "aA";
    string stones = "aAAbbbb";

    Solution sol;

    cout << sol.numJewelsInStones(jewels, stones) << endl;

    return 0;
}